crates/refunder/src/refund_service.rs

```diff
@@ -45,17 +48,20 @@ impl RefundService {
         ethflow_contracts: Vec<CoWSwapEthFlow::Instance>,
         min_validity_duration: i64,
         min_price_deviation_bps: i64,
-        account: Account,
+        signer: Box<dyn TxSigner<Signature> + Send + Sync + 'static>,
     ) -> Self {
+        let signer_address = signer.address();
+        let mut submitter_web3 = web3.clone();
+        submitter_web3.wallet.register_signer(signer);
```

Comment:
I just realized the API of `MutWallet` is a bit awkward. The reason is that they have to be cloned to add something to the 1 shared wallet state. This makes the code confusing here.

Instead we should make `register_signer` take a shared reference and do the clone internally. All this is not related to this PR and is not merge blocking. Just a thought that jumped out to me when I saw this code.